docs/shrc.md

```diff
@@ -285,6 +288,39 @@ The following APIs are available for obtaining information about a shared classe
 - `com.ibm.oti.shared.SharedClassUtilities`: Obtains detailed information about a shared classes cache, including its size, name, type, and status.
 - `com.ibm.oti.shared.SharedClassCacheInfo`: Stores information about a shared classes cache and provides API methods to retrieve the information and remove caches. You can also use the `IterateSharedCaches` and `DestroySharedCache` [JVMTI extensions](interface_jvmti.md#iteratesharedcaches).
 
+## Support for bytecode instrumentation
+
+Modifying the bytecode of a set of classes at runtime is a useful mechanism for adding functions to a program, such as profiling or debugging. The JVM Tools Interface (JVMTI) includes hooks that allow you to instrument the byte code in this way. Alternatively, you can write your own Java agent that uses the `java.lang.instrument` API. Sharing classes that are changed before they are loaded adds complexity to the class sharing process.
+
+By default, if OpenJ9 detects that a JVMTI agent or `java.lang.instrument` agent has registered to modify class bytes, modified classes are not stored in the cache. Instead, the VM stores original class byte data in the cache, which allows classes to be retransformed.
+
+If you turn off bytecode instrumentation support by specifying [`-Xshareclasses:disableBCI`](xshareclasses.md#disablebci), all bytecode is loaded from the file system for the agent to modify. When passed to the cache for storing, the VM compares the bytes with known classes of the same name as a safety measure. If a match is found, the class is reused. However, if a match is not found, the potentially modified class is stored in the cache in a way that prevents other VMs from loading it. In this situation, performance can be affected because the bytecode is always loaded from the file system. When bytecode instrumentation is turned off, classes cannot be retransformed.
```

Comment:
> If you turn off bytecode instrumentation support by specifying -Xshareclasses:disableBCI, all bytecode is loaded from the file system for the agent to modify.

This is true if you disable BCI support and do not use modification context (by default, there is no modification context). Not sure if it is easy for all user to infer this from the later paragraph talking about modification context.

Reply:
Thanks - I've updated this paragraph and linked to the sharing section.